src/components/common/NumberRoller/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef, useState } from 'react';
+import useInView from '@src/hooks/useInView';
+
+type NumberRollerProps = {
+  goalNumber: number;
+  rollRange?: number;
+};
+
+const NumberRoller = ({ goalNumber, rollRange = 50 }: NumberRollerProps) => {
+  const [number, setNumber] = useState(Math.max(goalNumber - rollRange, 0));
+  const wrapperRef = useRef<HTMLSpanElement>(null);
+  const { isInView } = useInView(wrapperRef, -20);
+
+  useEffect(() => {
+    const timerIntervalId = setInterval(() => {
+      if (isInView) {
+        setNumber((n) => {
+          if (n >= goalNumber) {
+            clearInterval(timerIntervalId);
+            return n;
+          }
+          return n + 1;
+        });
+      }
+    }, 100);
+    return () => {
+      clearInterval(timerIntervalId);
+    };
+  }, [goalNumber, isInView]);
```

Comment:
setInterval로 하여 제어가 좀 힘들다면, 아예 조건을 걸고 그 안에서 타이머를 거는 것은 어떨까요???

```suggestion
  useEffect(() => {
    if (isInView && number < goalNumber) {
      setTimeout(() => {
        setNumber((n) => n + 1);
      }, 100);
    }
  }, [goalNumber, isInView, number]);
```

Reply:
혹은 `intersectionObserver` API 에서, 뷰포트에 안 보이면(`if (!entry.isIntersecting)`) `setInterval`을 clear해주는 방식으로 해도 좋을 것 같아요 !!

Reply:
매우 훌륭한 방법입니다! number가 변하면 저 useEffect가 다시 돌아서 setTimeout이 계속 불리는 구조가 되겠군요 ..!!!! 너무너무 훌륭합니다~~!